content/rsf-spec/index.md

```diff
@@ -0,0 +1,263 @@
+pr:
+status: draft
+---
+
+# The Register Serialisation Format
+
+## Summary
+
+This RFC aims to collect in one place the current implementation of RSF so it
+can be added to the specification and it can be evolved with new RFCs when
+required.
+
+The Register Serialisation Format, from now on RSF, is an event log describing
+the evolution of the Register data and metadata.
+
+## Motivation
+
+TODO
+
+## Explanation
+
+### RSF Grammar
+
+RSF is a positional line-based textual format separated by tabs. Each
+line defines a command to apply to a Register state to obtain the next state.
+
+This specification uses the Augmented Backus-Naur Form (ABNF) as defined by
+[RFC5234](https://tools.ietf.org/html/rfc5234) and refined by
+[RFC7405](https://tools.ietf.org/html/rfc7405). It assumes the following
+definitions:
+
+* RFC5234: `ALPHA` (letters), `CRLF` (carriage return, line feed), `DIGIT`
+  (decimal digits), `HEXDIG` (hexadecimal digits) and `HTAB` (horizontal tab).
+* Registers specification: [`CANONREP`][canon-rep] (canonical representation).
+  Note that, in turn, it depends on [RFC8259](https://tools.ietf.org/html/rfc8259).
+
+```abnf
+log              = command *(CRLF command) [CRLF]
+command          = add-item / append-entry / assert-root-hash
+
+assert-root-hash = %s"assert-root-hash" HTAB hash
+
+add-item         = %s"add-item" HTAB CANONREP
+
+append-entry     = %s"append-entry" HTAB type HTAB key HTAB timestamp HTAB hash-list
+type             = "user" / "system"
+key              = alphanum / %x2D / %x5F
+hash-list        = hash *(list-separator hash)
+hash             = "sha-256:" 64(HEXDIG) ; sha-256
+list-separator   = ";" ; hash list separator
+
+alphanum         = ALPHA / DIGIT
+
+;                timestamp
+timestamp        = date "T" time
+date             = century year DSEP month DSEP day ; date YYYY-MM-DD
+time             = hour TSEP minute TSEP second TZ ; time HH:MM:SSZ
+
+;                date
+century          = 2DIGIT ; 00-99
+year             = 2DIGIT ; 00-99
+month            = 2DIGIT ; 01-12
+day              = 2DIGIT ; 01-28, 01-29, 01-30, 01-31 based on month/year
+DSEP             = "-"    ; date separator
+
+;                time
+hour             = 2DIGIT ; 00-24
+minute           = 2DIGIT ; 00-59
+second           = 2DIGIT ; 00-58, 00-59, 00-60 based on leap-second rules
+TSEP             = ":"    ; time separator
+TZ               = "Z"    ; timezone
+```
+
+### Media type
+
+The current media type is `application/uk-gov-rsf`. It should change to
+`application/vnd.rsf` to align with [RFC6838](https://tools.ietf.org/html/rfc6838).
+
+### REST API
+
+TODO
+
+The current implementation uses `GET /download-rsf`. The main issue with that
+is that diverges from the rest of the API where serialisation is expressed
+either via suffix or via media type. The problem with using the same approach,
+say `GET /register.rsf` is that we are not providing the same information when
+querying `GET /register.json`.
```

Comment:
Does this mean that the endpoint `GET /download-rsf` doesn't provide the same information as `GET /register.rsf`, or that the user has to provide different information?

```text
The current implementation uses `GET /download-rsf`. The main issue with that
is that diverges from the rest of the API where serialisation is expressed
either via suffix or via media type. The problem with using the same approach,
say `GET /register.rsf` is that we are not providing the same information when
querying `GET /register.json`.
```

Reply:
Well, `/register.json` right now offers some sort of summary-metadata of the register. RSF by nature is the full register. So both things are at odds.

Reply:
I didn't know about that endpoint :grimacing:.  It goes without saying that I think it should have a different name, e.g. `/summary.json`.

Reply:
At least it's clearer in intention 👍